lib/Analysis/OpFoldResultUtils.cpp

```diff
@@ -351,4 +371,50 @@ OpFoldResult compareOFRs(const OpFoldResult lhs, const OpFoldResult rhs,
   auto selectOp = b.create<arith::SelectOp>(loc, cmpOp, trueValue, falseValue);
   return selectOp.getResult();
 }
+
+OpFoldResult remOFRs(const OpFoldResult lhs, const OpFoldResult rhs,
+                     const Location loc, OpBuilder &b) {
+  auto lhsIntAttr = getIntAttr(lhs);
+  auto rhsIntAttr = getIntAttr(rhs);
+  // both lhs and rhs are constants, return result directly
+  if (lhsIntAttr && rhsIntAttr) {
+    if (rhsIntAttr.value() == 0) {
+      llvm_unreachable("Division by zero");
+    }
+    return b.getIndexAttr(lhsIntAttr.value() % rhsIntAttr.value());
+  }
+  // otherwise, need to create instructions to calculate new attribute value
+  auto lhsValue = dyn_cast<Value>(lhs);
+  if (lhsIntAttr) {
+    auto lhsOp =
+        b.create<arith::ConstantOp>(loc, b.getIndexAttr(lhsIntAttr.value()));
+    lhsValue = lhsOp.getResult();
+  }
+  auto rhsValue = dyn_cast<Value>(rhs);
+  if (rhsIntAttr) {
+    auto rhsOp =
+        b.create<arith::ConstantOp>(loc, b.getIndexAttr(rhsIntAttr.value()));
+    rhsValue = rhsOp.getResult();
```

Comment:
we are creating constant value with index type only to cast it back to int64 later. please create int64 value, perform the modulo, then finally cast the result to index

Reply:
Changed to i64.